@kiva/kv-shop/src/basketItems.ts

```diff
@@ -1,70 +1,76 @@
-import { gql, ApolloClient, NormalizedCacheObject } from '@apollo/client/core';
+import { gql, ApolloClient, DocumentNode } from '@apollo/client/core';
 import numeral from 'numeral';
-import { GraphQLErrors } from '@apollo/client/errors';
-import { getBasketID, hasBasketExpired, handleInvalidBasketForDonation } from './basket';
-import { parseShopError } from './shopError';
+import { getBasketID, hasBasketExpired, createBasket } from './basket';
+import { ShopError, parseShopError } from './shopError';
 
-export interface SetTipDonationOptions {
-	amount: string | number,
-	apollo: ApolloClient<NormalizedCacheObject>,
-}
-
-export async function setTipDonation({ amount, apollo }: SetTipDonationOptions) {
-	let data;
-	let error;
-	let hasFailedAddToBasket = false;
-	const donationAmount = numeral(amount).format('0.00');
+/**
+ * Call a shop mutation with a basketId, creating a new basket if necessary.
+ */
+async function callShopMutation(
+	apollo: ApolloClient<any>,
+	mutation: DocumentNode,
+	variables: Record<string, any>,
```

Comment:
GitHub linting is catching our usages of `any`. While not a terrible thing to catch, we could consider turning off that linting requirement to reduce noise in situations like `Record<string, any>` where it isn't as important to have a specific type.